client/my-sites/domains/domain-management/list/domain-row.jsx

```diff
@@ -148,9 +148,10 @@ class DomainRow extends PureComponent {
 	}
 
 	renderExpiryDate( expiryDate ) {
+		const { domain } = this.props;
 		return (
 			<div className="domain-row__registered-until-cell">
-				{ expiryDate ? expiryDate.format( 'LL' ) : '-' }
+				{ expiryDate && domain.type !== 'MAPPED' ? expiryDate.format( 'LL' ) : '-' }
```

Comment:
We have a constant defined for `'MAPPED'` in `client/lib/domains/constants.js`.  You can import it with something like `import { type as domainTypes } ...` and then use `domainTypes.MAPPED` here instead.